biorbd_optim/optimal_control_program.py

```diff
@@ -498,18 +498,22 @@ def solve(self, solver="ipopt", show_online_optim=False, options_ipopt={}):
         return solver.call(arg)
 
     def save(self, sol, file_path, to_numpy=False, **parameters):
-        _, ext = os.path.splitext(file_path)
-        if ext == "":
-            file_path = file_path + ".bo"
         dir, _ = os.path.split(file_path)
         if dir != "" and not os.path.isdir(dir):
             os.makedirs(dir)
 
-        with open(file_path, "wb") as file:
+        _, ext = os.path.splitext(file_path)
+        if to_numpy:
+            if ext == "":
```

Comment:
I think since it is now enforce to have .bob for numpy and .bo for normal, this should raise an error if not ".bob" or ""